examples/denoise_covar2d.py

```diff
@@ -75,23 +76,26 @@
 # Generate 2D clean images from input 3D map. The following statement can be used from the sim object:
 # imgs_clean = sim.clean_images(start=0, num=num_imgs)
 # To be consistent with the Matlab version in the numbers, we need to use the statements as below:
+logger.info('Generate random distributed rotation angles and obtain corresponding 2D clean images.')
 rots = qrand_rots(num_imgs, seed=0)
 imgs_clean = vol2img(vols[..., 0], rots)
 
 # Assign the CTF information and index for each image
 h_idx = np.array([filters.index(f) for f in sim.filters])
 
 # Evaluate CTF in the 8X8 FB basis
-h_ctf_fb = [radial_filter2fb_mat(filt.evaluate_k, ffbbasis) for filt in filters]
+h_ctf_fb = [radial_filter2fb_mat(filt.evaluate, ffbbasis) for filt in filters]
 
 # Apply the CTF to the clean images.
+logger.info('Apply CTF filters to clean images.')
 imgs_ctf_clean = Image(sim.eval_filters(imgs_clean))
 sim.cache(imgs_ctf_clean)
 
 # imgs_ctf_clean is an Image object. Convert to numpy array for subsequent statements
 imgs_ctf_clean = imgs_ctf_clean.asnumpy()
```

Comment:
This shouldn't be necessary. We should be able to add noise (or any stack of images) to an Image object.

Reply:
Agree. We should have simpler operation when the` Image` and `ImageSource` classes are finalized. The example is doing this way to make sure the added noises are consistent with the Matlab example.

Reply:
Ok. Let's make an issue.

Reply:
Open the issue #61 for addressing this in future.